bika/lims/browser/worksheet/views/analyses.py

```diff
@@ -71,216 +71,361 @@ def __init__(self, context, request):
-            else:
-                pos_text += "<td class='pos_top'>&nbsp;</td>"
-
-            pos_text += "<td class='pos_top_icons' rowspan='3'>"
-            if obj.portal_type == 'DuplicateAnalysis':
-                pos_text += "<img title='%s' src='%s/++resource++bika.lims.images/duplicate.png'/>" % (_("Duplicate").encode('utf-8'), self.context.absolute_url())
-                pos_text += "<br/>"
-            elif obj.portal_type == 'ReferenceAnalysis' and obj.ReferenceType == 'b':
-                pos_text += "<a href='%s'><img title='%s' src='++resource++bika.lims.images/blank.png'></a>" % (parent.absolute_url(), parent.Title())
-                pos_text += "<br/>"
-            elif obj.portal_type == 'ReferenceAnalysis' and obj.ReferenceType == 'c':
-                pos_text += "<a href='%s'><img title='%s' src='++resource++bika.lims.images/control.png'></a>" % (parent.absolute_url(), parent.Title())
-                pos_text += "<br/>"
-            if parent.portal_type == 'AnalysisRequest':
-                sample = parent.getSample()
-                pos_text += "<a href='%s'><img title='%s' src='++resource++bika.lims.images/sample.png'></a>" % (sample.absolute_url(), sample.Title())
-            pos_text += "</td></tr>"
-
-            pos_text += "<tr><td>"
-            if parent.portal_type == 'AnalysisRequest':
-                pos_text += "<a href='%s'>%s</a>" % (parent.absolute_url(), parent.Title())
-            elif parent.portal_type == 'ReferenceSample':
-                pos_text += "<a href='%s'>%s</a>" % (parent.absolute_url(), parent.Title())
-            elif obj.portal_type == 'DuplicateAnalysis':
-                pos_text += "<a style='white-space:nowrap' href='%s'>%s</a>" % (obj.getAnalysis().aq_parent.absolute_url(), obj.getReferenceAnalysesGroupID())
-            elif parent.portal_type == 'Worksheet':
-                parent = obj.getAnalysis().aq_parent
-                pos_text += "<a href='%s'>(%s)</a>" % (parent.absolute_url(), parent.Title())
-            pos_text += "</td></tr>"
-
-            # sampletype
-            pos_text += "<tr><td>"
-            if obj.portal_type == 'Analysis':
-                pos_text += obj.aq_parent.getSample().getSampleType().Title()
-            elif obj.portal_type == 'ReferenceAnalysis' or \
+    def get_slot_header(self, item):
+        """
+        Generates a slot header (the first cell of the row) for the item
+        :param item: the item for which the slot header is requested
+        :return: the html contents to be displayed in the first cell of a slot
+        """
+        obj = item['obj']
+        obj = api.get_object(obj)
+
+        # TODO All contents below have to be refactored/cleaned-up!
+        # fill the rowspan with a little table
+        # parent is either an AR, a Worksheet, or a
+        # ReferenceSample (analysis parent).
+        parent = obj.aq_parent
+        if parent.aq_parent.portal_type == "WorksheetFolder":
```

Comment:
You're using `portal_type` in the next coming checks more than once, so put it better into a variable, e.g. `portal_type = api.get_portal_type(parent)`